### ISSUE TITLE
Dropout vs keep probability

### ISSUE THREAD

**Issue by samadejacobs** [LLNL/lbann#1248]
Whereas LBANN uses keep probability in Dropout layer,[ Pytorch](https://pytorch.org/docs/stable/nn.html#dropout-layers) and [TensorFlow](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/keras/layers/Dropout) use dropout  probability (rate). I suggest we adopt Pytorch/TF2.0 approach in LBANN.
